pytorch_lightning/plugins/training_type/sharded.py

```diff
@@ -33,6 +36,7 @@ def configure_ddp(self):
         self._model = ShardedDataParallel(
             LightningShardedDataParallel(self.model), sharded_optimizer=self.lightning_module.trainer.optimizers
         )
+        setattr(self._model, "require_backward_grad_sync", False)
```

Comment:
nit: why `setattr` instead of `self._model.require_backward_grad_sync = False`

Reply:
because the model here is `ShardedDataParallel` and this attribute is only available for the `DistributedDataParallel`.
We check for this attribute in manual optmization.